sklearn/tree/tree.py

```diff
@@ -370,6 +370,31 @@ def predict(self, X):
             else:
                 return proba[:, :, 0]
 
+    def apply(self, X):
+        """
+        Returns the index of the leaf that each sample is predicted as.
```

Comment:
Maybe it should also say that it numbers all nodes, not only the leaves, and so the indices will go from 0 to ``2 ** (max_depth +1 ) -1`` but not all will be present?